qa/L0_https/test.sh

```diff
@@ -89,12 +98,67 @@ service nginx restart
 set +e
 
 # Test basic inference with https
-python $SIMPLE_INFER_CLIENT_PY -v -u localhost --ssl >> ${CLIENT_LOG}.ssl_infer 2>&1
+python $SIMPLE_INFER_CLIENT_PY -v -u localhost --ssl --keyfile client.key --certfile client.crt --cacerts ca.crt >> ${CLIENT_LOG}.ssl_infer 2>&1
 if [ $? -ne 0 ]; then
     cat ${CLIENT_LOG}.ssl_infer
     RET=1
 fi
 
+$TEST_CLIENT -v -u https://localhost:443 --keyfile client.key --certfile client.crt --cacerts ca.crt >> ${CLIENT_LOG}.c++.ssl_infer 2>&1
+if [ $? -ne 0 ]; then
+    cat ${CLIENT_LOG}.c++.ssl_infer
+    RET=1
+fi
+
+# Test basic inference on https without peer verification
+python $SIMPLE_INFER_CLIENT_PY -v -u localhost --ssl --insecure >> ${CLIENT_LOG}.ssl_infer_insecure 2>&1
+if [ $? -ne 0 ]; then
+    cat ${CLIENT_LOG}.ssl_infer_insecure
+    RET=1
+fi
+
+$TEST_CLIENT -v -u https://localhost:443 --verifyhost 0 --verifypeer 0 >> ${CLIENT_LOG}.c++.ssl_infer_insecure 2>&1
+if [ $? -ne 0 ]; then
+    cat ${CLIENT_LOG}.c++.ssl_infer_insecure
+    RET=1
+fi
+
+# Test failure cases for SSL
+# Try without SSL 
+$SIMPLE_INFER_CLIENT_PY -v -u localhost >> ${CLIENT_LOG}.no_ssl_fail_infer 2>&1
+if [ $? -ne 0 ]; then
```

Comment:
Kind of a small detail, but if I'm understanding this correctly, this test expects the exit code of line 128 to be 0, correct? And if it's not 0, then the test fails. But the comment and echo suggest that line 128 is supposed to fail--i.e. a non-zero exit code?

Reply:
Check condition checks if there is a failure, comments that the same was expected and if there is no failure sets RET to 1. Looks correct.

Reply:
oh, the `if` is the test success case, not the test failure case. the `else` is the test failure case. 😵‍💫